### PR TITLE
Add documentation for secretReferences

### PR DESCRIPTION
**Pull request by matt-simons**

### CODE REVIEW

File: docs/syncset.md [hunk 1]
```diff
@@ -41,6 +41,18 @@ spec:
     patch: |-
       { "data": { "foo": "new-bar" } }
     patchType: merge
+    
+  secretReferences:
+  - source:
+      kind: Secret
```
Comment (staebler) [inline docs/syncset.md]: The kind and apiVersion can be omitted in both the source and the target.

Comment (dgoodwin) [reply docs/syncset.md]: Yeah those fields would be unused here and they should probably be removed.

**Comment by dgoodwin** [comment]
Thanks a bunch for adding this @matt-simons

File: docs/syncset.md (hunk 1 above)
Comment (matt-simons) [reply docs/syncset.md]: cool, removed those

**Pull request merged**